packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -58,7 +63,47 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
         }
     }
 
-    onPrepare (config?: Options.Testrunner, capabilities?: Capabilities.RemoteCapabilities) {
+    async onPrepare (config?: Options.Testrunner, capabilities?: Capabilities.RemoteCapabilities) {
+        /**
+         * Upload app to BrowserStack if valid file path to app is given.
+         * Update app value of capability directly if app_url, custom_id, shareable_id is given
+         */
+        if (!this._options.app) {
+            log.info('app is not defined in browserstack-service config, skipping ...')
+        } else {
+            let app: App = {}
+            let appConfig: AppConfig | string = this._options.app
+
+            try {
+                app = await this._validateApp(appConfig)
+            } catch (error: unknown){
+                log.error(error)
+                return process.emit('SIGTERM')
+            }
+
+            if (['.apk', '.aab', '.ipa'].includes(path.extname(app.app!))){
```

Comment:
Let's move this (`['.apk', '.aab', '.ipa']`) to `constant.ts` for better readability.